Source/Scene/ModelExperimental/PointCloudStylingPipelineStage.js

```diff
@@ -0,0 +1,338 @@
+import AlphaMode from "../AlphaMode.js";
+import Cartesian3 from "../../Core/Cartesian3.js";
+import Cartesian4 from "../../Core/Cartesian4.js";
+import CesiumMath from "../../Core/Math.js";
+import Cesium3DTileRefine from "../Cesium3DTileRefine.js";
+import clone from "../../Core/clone.js";
+import defaultValue from "../../Core/defaultValue.js";
+import defined from "../../Core/defined.js";
+import Matrix4 from "../../Core/Matrix4.js";
+import ModelExperimentalType from "./ModelExperimentalType.js";
+import ModelExperimentalUtility from "./ModelExperimentalUtility.js";
+import OrthographicFrustum from "../../Core/OrthographicFrustum.js";
+import Pass from "../../Renderer/Pass.js";
+import PointCloudStylingStageVS from "../../Shaders/ModelExperimental/PointCloudStylingStageVS.js";
+import RuntimeError from "../../Core/RuntimeError.js";
+import SceneMode from "../SceneMode.js";
+import ShaderDestination from "../../Renderer/ShaderDestination.js";
+import VertexAttributeSemantic from "../VertexAttributeSemantic.js";
+
+const scratchUniform = new Cartesian4();
+
+/**
+ * The point cloud CPU styling stage is responsible for applying color,
+ * size, and show styles to point clouds at runtime. It also handles
+ * point cloud shading provided by either the model or the tileset that
+ * owns it, which is used if no style is provided.
```

Comment:
Need to read further to determine if "only if" is the correct wording here, but regardless I'd split out the last sentence for clarity.

```suggestion
 * owns it. Point cloud shading is applied only if no style is provided.
```